File: src/my_package/scripts/publisher.py
#!/usr/bin/env python
# license removed for brevity

import rospy 
from std_msgs.msg import Float32 
import random
def publisher() :
    pub  = rospy.Publisher("my_random_float_python",Float32,queue_size=10)
    rospy.init_node("publisher_python",anonymous= True) 
    rate =rospy.Rate(10) 
    random_float = Float32() 
    while not rospy.is_shutdown() :
        random_float.data = random.random()*10.0 
        pub.publish(data=random_float.data)
        rospy.loginfo("I am publishing {}".format(random_float.data)) 
        rate.sleep()



if __name__ == '__main__' :
    try :
        publisher() 
    except rospy.ROSInterruptException :
       pass 

    

File: src/my_package/src/publisher.cpp
#include "ros/ros.h"
#include<iostream>
#include<cstdlib>
#include "std_msgs/Float32.h"

int main(int argc ,char **argv){
	ros::init(argc,argv,"publisher") ;
	ros::NodeHandle publisher ;
	ros::Publisher pub = publisher.advertise<std_msgs::Float32>("my_random_float",1000) ;
	ros::Rate loop_rate(10) ;
	while(ros::ok()){
		std_msgs::Float32 random_float ;
	  random_float.data= (float)(std::rand()%100000)/10000.0 
		  ;
	  ROS_INFO("I am publishing %f",random_float.data) ;
	  pub.publish(random_float) ;
	  ros::spinOnce() ;
	  loop_rate.sleep() ;


	}
	return 0 ;

}
